---
title: "Single versus 10-strain inoculations of bacteria on Lemna"
author: "Jason Laurich and Megan Frederickson"
date:  "`r Sys.Date()`"
output: github_document
editor_options: 
  chunk_output_type: console
---

```{r Setup, include=FALSE}
library(knitr)
knit_hooks$set(crop = hook_pdfcrop)
knitr::opts_chunk$set(crop=TRUE)
```

This code generates the figures and model results for:

Laurich JR, Lash E, O'Brien AM, Pogoutse O, Frederickson ME. Community interactions among microbes give rise to host-microbiome mutualisms in an aquatic plant. 

### SECTION 1: Upload packages

```{r Packages, message=FALSE, warning=FALSE}
library(lme4) # mixed effects models
library(lmerTest) # get results of lmer models
library(car) #type 3 anovas
library(tidyverse) #manipulating data frames
library(cowplot) # luxury plotting
library(emmeans) # allows extraction of bacteria-specific regression estimates
library(ggrepel) # fine tune figures
library(ggtext) # fine tune figures
library(ggforce) # fine tune figures even more
library(kableExtra) # exports tables to LaTeX format
library(ape) #phylogenetics
library(phytools) #phylogenetics
library(caper) #phylogenetics
#if (!require("BiocManager", quietly = TRUE)) install.packages("BiocManager")
#BiocManager::install("ggtree")
library(ggtree) #luxury tree plotting
#devtools::install_github('datarootsio/artyfarty')
library(artyfarty) #ggplot themes
library(magick) #graphics
```

### SECTION 2: Upload data

```{r Data, message=FALSE, warning=FALSE}
#Load data
df.CW<-read.table('CW_mixed_inoculation_data.txt', header=F, na.strings="", sep="\t")
names(df.CW)<-c('pop','plt','bac','id','frd0','pix0','per0','grn0','grnper0','frd1','pix1','per1','grn1','grnper1','abs')

#Fix the structure for bacterial inoculation and plant
df.CW$bac <- factor(df.CW$bac, levels = c("Control", "1", "2", "3", "4", "5","6","7","8","9","10","All10"))
df.CW$plt <- as.factor(df.CW$plt)

# Add in edge effects. So for each 24 well plate, all but 6,7,10,11,14,15,18, and 19 are edge.
edge<-c(rep("Y",5), rep("N",2), rep("Y",2),rep("N",2), rep("Y",2), rep("N",2), rep("Y",2),rep("N",2), rep("Y",5))

#Final plate has 8 samples only
edge2<-rep(edge,38)
edge2<-c(edge2, rep("Y",5), rep("N",2), "Y")
df.CW$edge<-edge2

#Make edge a factor
df.CW$edge <- as.factor(df.CW$edge)

# Create vector for plate - 38 full, 39th with 8 wells
plate<-vector()
for (i in 1:38){
  plate<-c(plate, rep(i,24))
}
plate<-c(plate,rep(39,8))
df.CW$plate <- plate

#Makes plate number a factor
df.CW$plate <- as.factor(df.CW$plate)

#Take out errors (pipetting errors, wrong bacteria)
df.CW<-df.CW[-c(453,494,798),]

# Calculate growth.
df.CW$Grt<-df.CW$pix1 - df.CW$pix0

# Log abs count
df.CW$logabs<-log10(df.CW$abs)

#Let's create separate dataframes for each population 
df.C.2020<-subset(df.CW, df.CW$pop== "Churchill")
df.W.2020<-subset(df.CW, df.CW$pop== "Wellspring")

# Assign names from sequencing
#Churchill
df.C.2020$bac_name <- ifelse(df.C.2020$bac == 1, "Flavobacterium \nsp.", 
                             ifelse(df.C.2020$bac == 2, "Pseudomonas \nprotegens 1", 
                                    ifelse(df.C.2020$bac == 3, "Devosia \nconfluentis", 
                                           ifelse(df.C.2020$bac == 4, "Arcicella \nsp.", 
                                                  ifelse(df.C.2020$bac == 5, "Microbacterium \noxydans", 
                                                         ifelse(df.C.2020$bac == 6, "Aeromonas salmonicida", 
                                                                ifelse(df.C.2020$bac == 7, "Unidentified \nFlavobacteriaceae",
                                                                       ifelse(df.C.2020$bac == 8, "Bosea \nmassiliensis",                                                                                                 ifelse(df.C.2020$bac == 9, "Unidentified \nChitinophagaceae", 
                                                                                    ifelse(df.C.2020$bac == 10, "Falsiroseomonas sp.", as.character(df.C.2020$bac)))))))))))

#Wellspring
df.W.2020$bac_name <- as.factor(ifelse(df.W.2020$bac == 1, "Allorhizobium \nsp. 2", 
                             ifelse(df.W.2020$bac == 2, "Sphingomonas \npituitosa 1", 
                                    ifelse(df.W.2020$bac == 3, "Pseudomonas \nprotegens 2", 
                                           ifelse(df.W.2020$bac == 4, "Rhizobium sp.", 
                                                  ifelse(df.W.2020$bac == 5, "Rhizorhabdus \nwittichii 1", 
                                                         ifelse(df.W.2020$bac == 6, "Allorhizobium \nsp. 1", 
                                                                ifelse(df.W.2020$bac == 7, "Sphingomonas \npituitosa 2",
                                                                       ifelse(df.W.2020$bac == 8, "Unidentified \nHyphomicrobiales",                                                                                                 ifelse(df.W.2020$bac == 9, "Rhizorhabdus \nwittichi 2", ifelse(df.W.2020$bac == 10, "Rhizobium \nrosettiformans", as.character(df.W.2020$bac))))))))))))

#Bind them back together
df2020 <- rbind(df.C.2020, df.W.2020)

#Trim white space
df2020$pop <- trimws(df2020$pop)
```

### SECTION 3: Visualize phylogenetic tree

```{r Tree taxonomy, message=FALSE, warning=FALSE}
#Import tree that was created in QIIME2
tree <- ape::read.tree("tree.nwk")

#Import taxonomy generated by Qiime2 for Churchill and Wellspring field microbiomes
taxmat <- read.delim("church_well_taxonomy.tsv")

#List tree tip labels not in taxonomy from qiime2 (i.e., the single strains)
addtips <- tree$tip.label[which(!tree$tip.label %in% taxmat$Feature.ID)]

#Add these tips to taxonomy
addtax <- data.frame("Feature.ID" = addtips, "Taxon" = NA, "Confidence" = NA)
taxmat <- rbind(taxmat, addtax)

#Split taxonomy across columns
ranks <- c("domain","phylum","class","order","family","genus","species")

#Clean up and format dataframe
taxonomy <- taxmat %>%
  mutate_at('Taxon',str_replace_all, "[a-z]__","") %>%
  separate(Taxon, sep = ';', into=ranks,remove = TRUE) %>%
  column_to_rownames(var = "Feature.ID") %>%
  as.matrix()

#Write file to disk
write.csv(taxonomy, file="merged_taxonomy.csv")
```

For the next step, we manually input the single strain taxonomy information into csv file, and then uploaded it again. 

```{r Update taxonomy, warning=FALSE, message=FALSE}
#Read back in the edited taxonomy file
taxonomy <- read.csv("merged_taxonomy_edited.csv")

#Classify tips as field asvs or cultured isolates
taxonomy$Type <- ifelse(taxonomy$label %in% addtips, "Cultured isolate", "Field ASV")

#Label the cultured isolates by species name
taxonomy$tree_label <- ifelse(taxonomy$Type == "Cultured isolate", taxonomy$species, "")

#Read in feature table
featuretable <- read.csv("church_well_feature_table.csv")

#List tree tip labels not in feature table from qiime2 (i.e., the single strains)
addtips <- tree$tip.label[which(!tree$tip.label %in% featuretable$Feature.ID)]

#Add these tips to feature table
addtax <- data.frame("Feature.ID" = addtips, "Church.F.28" = NA, "Well.F.92" = NA)
addtax$Church.F.28 <- ifelse(substring(addtax$Feature.ID, 1, 1) == "C", 1,0)
addtax$Well.F.92 <- ifelse(substring(addtax$Feature.ID, 1, 1) == "W", 1,0)
featuretable <- rbind(featuretable, addtax)
colnames(featuretable) <- c("label", "Churchill", "Wellspring")

#Add single strains to feature table
taxonomy <- merge(taxonomy, featuretable, by="label")
taxonomy$Population <- ifelse(taxonomy$Churchill > 0 & taxonomy$Wellspring == 0, "Churchill", ifelse(taxonomy$Churchill == 0 & taxonomy$Wellspring > 0, "Wellspring", "Both"))

#Identify sisters, etc.
taxonomy$sister_clade <- sapply(taxonomy$label, getSisters, tree=tree)
taxonomy$sibs <- sapply(taxonomy$sister_clade, getDescendants, tree=tree)
taxonomy$tree_class <- ifelse(taxonomy$Type == "Cultured isolate", "", taxonomy$class)
taxonomy$tree_order <- ifelse(taxonomy$Type == "Cultured isolate", "", taxonomy$order)
taxonomy$tree_family <- ifelse(taxonomy$Type == "Cultured isolate", "", taxonomy$family)
taxonomy$tree_genus <- ifelse(taxonomy$Type == "Cultured isolate", "", taxonomy$genus)
taxonomy$tree_asvlabel <- ifelse(taxonomy$Type == "Cultured isolate", "", ifelse(taxonomy$genus != "", taxonomy$genus, taxonomy$family))
```

Use ggtree to plot the tree. 

```{r Plot tree, warning=FALSE, message=FALSE}
#Make type and population factors
taxonomy$Type <- as.factor(taxonomy$Type)
taxonomy$Population <- as.factor(taxonomy$Population)

p <- ggtree(tree, layout="circular") %<+% taxonomy + 
  geom_tippoint(aes(shape = Type, color = Population), size=1.5, alpha=0.8)+ 
  theme(legend.direction="horizontal", legend.position=c(0.5,0.1), legend.box = "horizontal")+
  geom_tiplab2(aes(label=trimws(tree_asvlabel)), size=1.5, offset=0.005)+
  geom_tiplab2(aes(label=trimws(tree_label)), size=1.5, offset=0.005, color = "blue")+
  guides(color = guide_legend(override.aes = list(size = 3)), shape=guide_legend(override.aes = list(size=3)))

#p <- p + geom_text(aes(label=node), size=0.8, vjust=-2) #This visualizes node numbers, if needed

pdf("Figure_2.pdf", width=8, height=9)
p
dev.off()
```

### SECTION 3: Microbial growth models and figures

First, let's fit a model combining all data (with bacterial strain nested within population as a random effect) in which we test whether the productivity of 10-strain communities differs from the productivity of single strains. Again, this tests whether there is a biodiversity-ecosystem function relationship, whereby the number of strains in the microbiome (1 or 10) predicts microbial productivity. 

```{r Microbial growth models both pops, message=FALSE, warnings=FALSE}
#Group by whether 0, 1, or 10 strains were added
df2020$group <- as.factor(ifelse(df2020$bac == "All10", "10-strain community", ifelse(df2020$bac == "Control", "Control", "Single strain")))

#Adjust label names
df2020[df2020$group == "10-strain community", "bac_name"] <- "All 10 bacteria"

#Make single strain the reference
df2020 = df2020 %>% mutate(group = relevel(group, "Single strain"))

#hist(log(df2020$abs))
df2020$concat <- paste0(df2020$pop, " ", df2020$bac_name) #Concatenate pop and bac names, since there are different bacteria in each population
df2020$group <- as.factor(df2020$group)
df2020$concat <- as.factor(df2020$concat)
lmer.abs <-lmer(log(as.numeric(abs))~group*plt+(1|edge)+(1|plate)+(1|concat), data=subset(df2020, bac != "Control"))
summary(lmer.abs)
#Anova(lmer.abs, type=3)
ranova(lmer.abs) #Random effects
#plot(lmer.abs) #Looks okay
```

Generate LaTeX Table 1 for the manuscript.  
 
```{r Latex tables, message=FALSE, warnings = FALSE}
tbl <- as.data.frame(coef(summary(lmer.abs)))
tbl[, 1:4] <- round(tbl[ , 1:4], 3)
colnames(tbl) <- c("Estimate", "SE", "df", "t-value", "p-value")
rownames(tbl) <- c("Intercept", "10-strain community", "Host", "10-strain community x host")
tbl$`p-value` <- ifelse(tbl$`p-value` < 0.001, "< 0.001", tbl$`p-value`)

kbl(tbl, caption = "Model results for the effect of microbial strain diversity (one versus ten strains) on microbial productivity. Intercept is a single microbial strain growing in the absence of a host. Estimates are on a log scale.", booktabs = T, format = "latex") %>% kable_styling(position = "center")
```

Next, let's calculate the additive expectation for productivity in 10-strain communities, given the productivity of each single strain grown separately.

```{r Microbial growth mean and additive predictions,message=FALSE, warnings=FALSE}
#Make the 10-strain treatment the reference
df2020$bac_name <- as.factor(df2020$bac_name)
df2020 = df2020 %>% mutate(bac_name = relevel(bac_name, "All 10 bacteria"))

#Churchill 2020
#Model 
lmer.abs.C<-lmer(abs~bac_name*plt + (1|edge) + (1|plate), data=subset(df2020, bac_name != "Control" & pop=="Churchill"))
summary(lmer.abs.C)
#plot(lmer.abs.C) # Not great, but easier/more interpretable to leave this on a raw scale for estimating additive expectation, rather than log-transforming response variable

#Additive and mean predictions
C_emmeans <- as.data.frame(emmeans(lmer.abs.C, specs = ~bac_name+plt))
C_micr_mean_noplt <- mean(C_emmeans[C_emmeans$plt == "N" & C_emmeans$bac_name != "All 10 bacteria", "emmean"])
C_micr_mean_plt <- mean(C_emmeans[C_emmeans$plt == "Y" & C_emmeans$bac_name != "All 10 bacteria", "emmean"])
C_micr_sum_noplt <- sum(C_emmeans[C_emmeans$plt == "N" & C_emmeans$bac_name != "All 10 bacteria", "emmean"])
C_micr_sum_plt <- sum(C_emmeans[C_emmeans$plt == "Y" & C_emmeans$bac_name != "All 10 bacteria", "emmean"])

#Wellspring 2020

#Model 
lmer.abs.W<-lmer(abs~bac_name*plt + (1|edge) + (1|plate), data=subset(df2020, bac_name != "Control" & pop=="Wellspring"))
summary(lmer.abs.W)

#Additive and mean predictions
W_emmeans <- as.data.frame(emmeans(lmer.abs.W, specs = ~bac_name+plt))
W_micr_mean_noplt <- mean(W_emmeans[W_emmeans$plt == "N" & W_emmeans$bac_name != "All 10 bacteria", "emmean"])
W_micr_mean_plt <- mean(W_emmeans[W_emmeans$plt == "Y" & W_emmeans$bac_name != "All 10 bacteria", "emmean"])
W_micr_sum_noplt <- sum(W_emmeans[W_emmeans$plt == "N" & W_emmeans$bac_name != "All 10 bacteria", "emmean"])
W_micr_sum_plt <- sum(W_emmeans[W_emmeans$plt == "Y" & W_emmeans$bac_name != "All 10 bacteria", "emmean"])
```

Next, let's make a figure of the microbial data. 

```{r Microbial growth figures, warnings=FALSE, message=FALSE, fig.height=8, fig.width=8}

#Set figure defaults
pt_size = 3
ylimits = c(0, 50000)
zero_line <- "red3"
sum_line <- "green3"
mean_line <- "blue3"
dot_colors <- c("blue", "red", "black")

# Generate Figure 3
Sum_2020_C_micr <- subset(df2020, pop == "Churchill") %>% group_by(bac_name, plt) %>% summarize(n=n(), mean=mean(abs, na.rm=TRUE),
sd = sd(abs, na.rm=TRUE), sem = sd/sqrt(n))

#Add colors
Sum_2020_C_micr$color <- ifelse(Sum_2020_C_micr$bac_name == "All10", "10-strain community", ifelse(Sum_2020_C_micr$bac_name == "Control", "Control", "Single strain"))

#Churchill WITHOUT PLANTS
plot_C2020_micr_noplt <- ggplot(subset(Sum_2020_C_micr, bac_name != "Control" & plt == "N"), aes(y=mean,x=reorder(bac_name, mean), color=color))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name, mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  xlab("Bacterial treatment")+
  ggtitle("Churchill, no plants")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
  theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.1, 0.9))+
  geom_hline(yintercept=C_micr_mean_noplt, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=C_micr_sum_noplt, linetype="dashed", color = sum_line,linewidth=1)+
  ylab(expression("Microbial density (cells/µL)"))

plot_C2020_micr_noplt_updated <- plot_C2020_micr_noplt+
   scale_x_discrete(labels=c("*Microbacterium oxydans*", "*Bosea massiliensis*",  "*Flavobacterium* sp.", "*Falsiroseomonas* sp.", "*Devosia confluentis*", "Unidentified *Flavobacteriaceae*", "*Aeromonas salmonicida*", "*Arcicella* sp.", "Unidentified *Chitinophagaceae*", "All 10 bacteria", "*Pseudomonas protogens* 1"))+
  theme(axis.text.x = ggtext::element_markdown())

plot_C2020_micr_noplt_updated_v2 <-plot_C2020_micr_noplt_updated+facet_zoom(ylim = c(0, 10000))

#Churchill WITH PLANTS
plot_C2020_micr_plt <- ggplot(subset(Sum_2020_C_micr, bac_name != "Control" & plt == "Y"), aes(y=mean,x=reorder(bac_name, mean), color=color))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name, mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  xlab("Bacterial treatment")+
  ggtitle("Churchill, with plants")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
  theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.1, 0.9))+
  geom_hline(yintercept=C_micr_mean_plt, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=C_micr_sum_plt, linetype="dashed", color = sum_line,linewidth=1)+
  ylab(expression("Microbial density (cells/µL)"))

plot_C2020_micr_plt_updated <- plot_C2020_micr_plt+
   scale_x_discrete(labels=c("*Devosia confluentis*", "*Bosea massiliensis*", "*Aeromonas salmonicida*", "*Microbacterium oxydans*",  "*Arcicella* sp.", "*Falsiroseomonas* sp.", "*Flavobacterium* sp.",  "Unidentified *Flavobacteriaceae*", "Unidentified *Chitinophagaceae*", "All 10 bacteria", "*Pseudomonas protogens* 1"))+
  theme(axis.text.x = ggtext::element_markdown())

plot_C2020_micr_plt_updated_v2 <-plot_C2020_micr_plt_updated+facet_zoom(ylim = c(0, 10000))

# Wellspring
Sum_2020_W_micr <- subset(df2020, pop=="Wellspring") %>% group_by(bac_name, plt) %>% summarize(n=n(), mean=mean(abs, na.rm=TRUE),
sd = sd(abs, na.rm=TRUE), sem = sd/sqrt(n))

#Add colors
Sum_2020_W_micr$color <- ifelse(Sum_2020_W_micr$bac_name == "All10", "10-strain community", ifelse(Sum_2020_W_micr$bac_name == "Control", "Control", "Single strain"))

#Wellspring WITHOUT PLANTS
plot_W2020_micr_noplt <- ggplot(subset(Sum_2020_W_micr, bac_name != "Control" & plt == "N"), aes(y=mean,x=reorder(bac_name, mean), color=color))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name, mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  xlab("Bacterial treatment")+
  ggtitle("Wellspring, no plants")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
  theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.1, 0.9))+
  geom_hline(yintercept=W_micr_mean_noplt, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=W_micr_sum_noplt, linetype="dashed", color = sum_line,linewidth=1)+
  ylab(expression("Microbial density (cells/µL)"))

plot_W2020_micr_noplt_updated <- plot_W2020_micr_noplt+
   scale_x_discrete(labels=c("Unidentified *Hyphomicrobiales*", "*Rhizorhabdus wittichii* 1", "*Rhizorhabdus wittichii* 2", "*Rhizobium* sp.", "*Allorhizobium* sp. 1",  "*Sphingomonas pituitosa* 2", "*Sphingomonas pituitosa* 1", "*Rhizobium rosettiformans*", "*Allorhizobium* sp. 2", "*Pseudomonas protogens* 2", "All 10 bacteria"))+
  theme(axis.text.x = ggtext::element_markdown())

plot_W2020_micr_noplt_updated_v2 <-plot_W2020_micr_noplt_updated+facet_zoom(ylim = c(0, 8000))

#Wellspring WITH PLANTS
plot_W2020_micr_plt <- ggplot(subset(Sum_2020_W_micr, bac_name != "Control" & plt == "Y"), aes(y=mean,x=reorder(bac_name, mean), color=color))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name, mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  xlab("Bacterial treatment")+
  ggtitle("Wellspring, with plants")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
  theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.1, 0.9))+
  geom_hline(yintercept=W_micr_mean_plt, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=W_micr_sum_plt, linetype="dashed", color = sum_line,linewidth=1)+
  ylab(expression("Microbial density (cells/µL)"))

plot_W2020_micr_plt_updated <- plot_W2020_micr_plt+
   scale_x_discrete(labels=c("*Allorhizobium* sp. 1", "Unidentified *Hyphomicrobiales*", "*Sphingomonas pituitosa* 1", "*Rhizorhabdus wittichii* 1", "*Allorhizobium* sp. 2", "*Rhizorhabdus wittichii* 2", "*Pseudomonas protogens* 2", "*Rhizobium rosettiformans*", "*Rhizobium* sp.", "All 10 bacteria", "*Sphingomonas pituitosa* 2"))+
  theme(axis.text.x = ggtext::element_markdown())

plot_W2020_micr_plt_updated_v2 <-plot_W2020_micr_plt_updated+facet_zoom(ylim = c(0, 8000))

final_microbes <- plot_grid(plot_C2020_micr_noplt_updated_v2, plot_C2020_micr_plt_updated_v2,plot_W2020_micr_noplt_updated_v2, plot_W2020_micr_plt_updated_v2, nrow=2, labels="AUTO")
save_plot("Figure_3.pdf", final_microbes, base_height=12, base_width=18)
final_microbes
```

Next, we want to plot the strain and communities means without a host (x-axis) versus with a host (y-axis). 

```{r Host versus no host, message=FALSE, warning=FALSE}
Sum_2020_C_micr_wide <- Sum_2020_C_micr %>% pivot_wider(names_from = plt, values_from = n:sem)

lm_micr_C <- lm(log(mean_Y)~log(mean_N), data=Sum_2020_C_micr_wide)
summary(lm_micr_C)

C_alt <-ggplot(subset(Sum_2020_C_micr_wide, bac_name != "Control"))+geom_point(aes(x=mean_N, y=mean_Y, color=color))+geom_abline(aes(intercept=0, slope=1), linetype="dotted")+theme_cowplot()+xlab(expression("Microbial density without hosts (cells/µL)"))+ylab(expression("Microbial density with hosts (cells/µL)"))+
  geom_errorbar(aes(x=mean_N, ymax=mean_Y+sem_Y, ymin=mean_Y-sem_Y, color=color))+
    geom_errorbarh(aes(xmax=mean_N+sem_N, xmin=mean_N-sem_N, y=mean_Y, color=color))+
 theme(legend.title = element_blank())+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.55, 0.15))+
    ggtitle("Churchill")+
    geom_text_repel(data=subset(Sum_2020_C_micr_wide, bac_name != "Control" & bac_name !="All 10 bacteria"), aes(x=mean_N, y=mean_Y, label=trimws(gsub("[\r\n]", "", bac_name))),fontface = "italic", size=2.5)+
    geom_text_repel(data=subset(Sum_2020_C_micr_wide, bac_name == "All 10 bacteria"), aes(x=mean_N, y=mean_Y, label=trimws(gsub("[\r\n]", "", bac_name))), size=2.5)+
    scale_y_log10(limits=c(10,10000))+  
  scale_x_log10(limits=c(10,10000))+
  geom_smooth(aes(x=mean_N, y=mean_Y), method="lm", se=FALSE)
C_alt

Sum_2020_W_micr_wide <- Sum_2020_W_micr %>% pivot_wider(names_from = plt, values_from = n:sem)

lm_micr_W <- lm(log(mean_Y)~log(mean_N), data=Sum_2020_W_micr_wide)
summary(lm_micr_W)

W_alt <-ggplot(subset(Sum_2020_W_micr_wide, bac_name != "Control"))+geom_point(aes(x=mean_N, y=mean_Y, color=color))+geom_abline(aes(intercept=0, slope=1), linetype="dotted")+theme_cowplot()+xlab(expression("Microbial density without hosts (cells/µL)"))+ylab(expression("Microbial density with hosts (cells/µL)"))+
 theme(legend.title = element_blank())+
    geom_errorbar(aes(x=mean_N, ymax=mean_Y+sem_Y, ymin=mean_Y-sem_Y, color=color))+
    geom_errorbarh(aes(xmax=mean_N+sem_N, xmin=mean_N-sem_N, y=mean_Y, color=color))+
  scale_color_manual(values=c(dot_colors[1], dot_colors[3]))+
  theme(legend.position = c(0.55, 0.15))+
    ggtitle("Wellspring")+
  geom_text_repel(data=subset(Sum_2020_W_micr_wide, bac_name != "Control" & bac_name !="All 10 bacteria"), aes(x=mean_N, y=mean_Y, label=trimws(gsub("[\r\n]", "", bac_name))),fontface = "italic", size=2.5)+
    geom_text_repel(data=subset(Sum_2020_W_micr_wide, bac_name == "All 10 bacteria"), aes(x=mean_N, y=mean_Y, label=trimws(gsub("[\r\n]", "", bac_name))), size=2.5)+
    scale_y_log10(limits=c(10,10000))+  
  scale_x_log10(limits=c(10,10000))+
  geom_smooth(aes(x=mean_N, y=mean_Y), method="lm", se=FALSE)
W_alt

alt <- plot_grid(C_alt, W_alt, labels="AUTO")
alt

save_plot("Figure_4.pdf", alt, base_height=6, base_width=12)

```

### SECTION 4: Plant growth models and figures

The first set of models tests whether there is a biodiversity-ecosystem function relationship, whereby the number of strains in the microbiome (0, 1, or 10) predicts plant growth. 

```{r Duckweed growth models both pops, message=FALSE, warning=FALSE}
#Make the control (uninoculated) treatment the reference
df2020 = df2020 %>% mutate(group = relevel(group, "Single strain"))
df2020 = df2020 %>% mutate(group = relevel(group, "Control"))

#Model both populations together, with bacterial treatment nested in population as a random effect
#Frond number
#hist(df2020$frd1)
lm_both_frd <- lmer(frd1~frd0+group+(1|edge)+(1|plate)+(1|pop/bac_name), data=df2020)
summary(lm_both_frd)
Anova(lm_both_frd, type=3)
emmeans(lm_both_frd, list(pairwise ~ group), adjust="tukey")
#plot(lm_both_frd) #Looks ok

#Pixel area
#hist(df2020$pix1)
lm_both_pix <- lmer(pix1~scale(pix0)+group+(1|edge)+(1|plate)+(1|pop/bac_name), data=df2020)
summary(lm_both_pix)
Anova(lm_both_pix, type=3)
ranova(lm_both_pix)
emmeans(lm_both_pix, list(pairwise ~ group), adjust="tukey")
#plot(lm_both_pix) #Looks ok
```

The second set of models fits each population separately and each bacterium as a fixed effect to calculate additive expectation for 10-strain community from single-strain inoculations. 

```{r Duckweed growth models, message=FALSE, warning=FALSE}
#Make the control treatment the reference
df2020 = df2020 %>% mutate(bac_name = relevel(bac_name, "Control"))

#Churchill
C_pixels <- lmer(pix1~pix0+bac_name+(1|edge)+(1|plate), data=subset(df2020, pop == "Churchill"))
summary(C_pixels)
#plot(C_pixels) #Looks ok

#Calculate additive and mean predictions 
C_pix_emmeans <- as.data.frame(emmeans(C_pixels, specs=~bac_name))
pred.C.2020.avg.emmeans <- mean(as.numeric(C_pix_emmeans[3:12, "emmean"])) 
pred.C.2020.sum.emmeans <- sum(as.numeric(C_pix_emmeans[3:12, "emmean"]-as.numeric(C_pix_emmeans[1, "emmean"])))+as.numeric(C_pix_emmeans[1, "emmean"]) #Add the difference between the mean for each single strain and the control, and then add the control

#Wellspring
W_pixels <- lmer(pix1~pix0+bac_name+(1|edge)+(1|plate), data=subset(df2020, pop == "Wellspring"))
summary(W_pixels)
#plot(W_pixels) #Looks ok

#Calculate additive and mean predictions 
#emmeans
W_pix_emmeans <- as.data.frame(emmeans(W_pixels, specs=~bac_name))
pred.W.2020.avg.emmeans <- mean(as.numeric(W_pix_emmeans[3:12, "emmean"]))
pred.W.2020.sum.emmeans <- sum(as.numeric(W_pix_emmeans[3:12, "emmean"]-as.numeric(W_pix_emmeans[1, "emmean"])))+as.numeric(W_pix_emmeans[1, "emmean"]) 
```

Next, let's make a figure of the plant growth data. 

```{r Duckweed growth figures, message=FALSE, warnings=FALSE}
pt_size = 3
ylimits = c(12000, 48000)
zero_line <- "red3"
sum_line <- "green3"
mean_line <- "blue3"
dot_colors <- c("red","black", "blue")

Sum_2020 <- df2020 %>% group_by(pop, bac_name, group) %>% summarize(n=n(), mean=mean(pix1, na.rm=TRUE), sd=sd(pix1, na.rm=TRUE), sem=sd/sqrt(n))

plot_C2020 <- ggplot(subset(Sum_2020, pop == "Churchill"), aes(y=mean,x=reorder(bac_name, mean), color=group))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name,mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  geom_hline(yintercept=pred.C.2020.avg.emmeans, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=pred.C.2020.sum.emmeans, linetype="dashed", color = sum_line,linewidth=1)+
  xlab("Bacterial treatment")+
  ggtitle("Churchill")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
   theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=dot_colors)+
  theme(legend.position = c(0.1, 0.9))+
  ylab("Duckweed area (final pixels)")

plot_C2020_updated <- plot_C2020+scale_x_discrete(labels=c("*Flavobacterium* sp.", "*Bosea massiliensis*", "Control", "Unidentified *Flavobacteriaceae*", "*Aeromonas salmonicida*", "*Devosia confluentis*", "*Falsiroseomonas* sp.", "Unidentified *Chitinophagaceae*", "*Arcicella* sp.", "*Microbacterium oxydans*", "All 10 bacteria",  "*Pseudomonas protegens*"))+
  theme(axis.text.x = ggtext::element_markdown())
plot_C2020_updated

#Wellspring 2020
plot_W2020 <- ggplot(subset(Sum_2020, pop == "Wellspring"), aes(y=mean,x=reorder(bac_name, mean), color=group))+
  geom_point(size=pt_size)+
  geom_errorbar(aes(x=reorder(bac_name,mean), ymin=mean-sem,ymax=mean+sem),width=0.5)+
  theme_cowplot()+
  geom_hline(yintercept=pred.W.2020.avg.emmeans, linetype="dashed", color = mean_line,linewidth=1)+
  geom_hline(yintercept=pred.W.2020.sum.emmeans, linetype="dashed", color = sum_line, linewidth=1)+
  xlab("Bacterial treatment")+
  ggtitle("Wellspring")+
  theme(axis.text.x=element_text(angle=90, vjust=0.5, hjust=1))+
  theme(legend.title = element_blank())+
  scale_y_continuous(limits=ylimits)+
  scale_color_manual(values=dot_colors)+
  theme(legend.position="none")+
   ylab("Duckweed area (final pixels)")

plot_W2020_updated <- plot_W2020+scale_x_discrete(labels=c("*Sphingomonas pituitosa* 1", "Control",  "Unidentified *Hyphomicrobiales*", "*Allorhizobium* sp. 2", "*Allorhizobium* sp. 1", "*Rhizorhabdus wittichii* 2", "*Pseudomonas protegens* 2", "*Rhizorhabdus wittichii* 1", "*Rhizobium* sp.", "*Sphingomonas pituitosa* 2", "All 10 bacteria", "*Rhizobium rosettiformans*"))+
  theme(axis.text.x = ggtext::element_markdown())
plot_W2020_updated

plot_2020 <- plot_grid(plot_C2020_updated, plot_W2020_updated, labels="AUTO", align='h')
plot_2020
save_plot("Figure_5.pdf", plot_2020, base_height=8, base_width=12)
```

### SECTION 5: Fitness regression

Fit models (with full random effect structure) from which we can extract 'genotype' (i.e. bacterial inocula) means for their (1) effects on duckweed growth, and (2) microbial density when grown with plants. Then, we will use scaled data so that we can center fitness proxy data around 0.

```{r Fitness regression, warning=FALSE, message=FALSE}
#Churchill
df.C.2020 <- subset(df2020, bac_name != "Control" & plt == "Y" & pop == "Churchill")
df.C.2020$pix1_scaled <- scale(df.C.2020$pix1, center = TRUE, scale = TRUE)
df.C.2020$pix0_scaled <- scale(df.C.2020$pix0, center = TRUE, scale = TRUE)
df.C.2020$abs_scaled <- scale(df.C.2020$abs, center=TRUE, scale=TRUE)

# Create models to extract strain/genotype means
modC <-lmer(pix1_scaled ~ bac_name + pix0_scaled + (1|edge) +(1|plate), data=df.C.2020)
emC <- as.data.frame(emmeans(modC, "bac_name", var="pix1_scaled"))
emC$CI<-emC$SE*qnorm(0.975)

modC.abs<-lmer(abs_scaled ~ bac_name + (1|edge) +(1|plate), data=df.C.2020)
em.C.abs<-as.data.frame(emmeans(modC.abs, "bac_name", var="abs"))

emC$color <- ifelse(emC$bac_name == "All 10 bacteria", "10-strain community", "Single strain")

emC$abs<-em.C.abs$emmean
emC$abs.SE<-em.C.abs$SE

# Is there a significant correlation between bacterial cell density and duckweed growth?
regC<-lm(emmean~abs,data=emC)
summary(regC)

plot_C_fitness <- ggplot(emC)+
  annotate("rect", xmin = 0, xmax = Inf, ymin = 0, ymax = Inf, 
           alpha = .2)+
  geom_vline(xintercept=0, linetype="dotted")+
  geom_hline(yintercept=0, linetype="dotted")+
  geom_point(aes(x=abs, y=emmean, color=color))+theme_cowplot()+xlab(expression("Microbial density (cells/µL, centered and scaled)"))+ylab(expression("Duckweed area (pixels, centered and scaled)"))+
#geom_errorbar(aes(x=abs, ymax=emmean+SE, ymin=emmean-SE, color=color))+
#geom_errorbarh(aes(xmax=abs+abs.SE, xmin=abs-abs.SE, y=emmean, color=color))+
 theme(legend.title = element_blank())+
  scale_color_manual(values=c(dot_colors[3], dot_colors[2]))+
  theme(legend.position = c(0.55, 0.15))+
  geom_text_repel(data=subset(emC, bac_name != "All 10 bacteria"), aes(x=abs, y=emmean, label=trimws(gsub("[\r\n]", "", bac_name))),fontface = "italic", size=2.5)+
      geom_text_repel(data=subset(emC, bac_name == "All 10 bacteria"), aes(x=abs, y=emmean, label=trimws(gsub("[\r\n]", "", bac_name))), size=2.5)+
  geom_smooth(aes(x=abs, y=emmean), method="lm", se=FALSE)+
    scale_y_continuous(limits=c(-0.6,0.6))+
  scale_x_continuous(limits=c(-2,2.5))+
      ggtitle("Churchill")
plot_C_fitness

#Wellspring
df.W.2020 <- subset(df2020, bac_name != "Control" & plt == "Y" & pop == "Wellspring")
df.W.2020$pix1_scaled <- scale(df.W.2020$pix1, center = TRUE, scale = TRUE)
df.W.2020$pix0_scaled <- scale(df.W.2020$pix0, center = TRUE, scale = TRUE)
df.W.2020$abs_scaled <- scale(df.W.2020$abs, center=TRUE, scale=TRUE)

# Create models to extract strain/genotype means
modW <-lmer(pix1_scaled ~ bac_name + pix0_scaled + (1|edge) +(1|plate), data=df.W.2020)
emW <- as.data.frame(emmeans(modW, "bac_name", var="pix1_scaled"))
emW$CI<-emW$SE*qnorm(0.975)

modW.abs<-lmer(abs_scaled ~ bac_name + (1|edge) +(1|plate), data=df.W.2020)
em.W.abs<-as.data.frame(emmeans(modW.abs, "bac_name", var="abs"))

emW$color <- ifelse(emW$bac_name == "All 10 bacteria", "10-strain community", "Single strain")

emW$abs<-em.W.abs$emmean
emW$abs.SE<-em.W.abs$SE

# Is there a significant correlation between bacterial cell density and duckweed growth?
regW<-lm(emmean~abs,data=emW)
summary(regW)

plot_W_fitness <- ggplot(emW)+annotate("rect", xmin = 0, xmax = Inf, ymin = 0, ymax = Inf, 
           alpha = .2)+
  geom_vline(xintercept=0, linetype="dotted")+
  geom_hline(yintercept=0, linetype="dotted")+
geom_point(aes(x=abs, y=emmean, color=color))+theme_cowplot()+xlab(expression("Microbial density (cells/µL, centered and scaled)"))+ylab(expression("Duckweed area (pixels, centered and scaled)"))+
#geom_errorbar(aes(x=abs, ymax=emmean+SE, ymin=emmean-SE, color=color))+
#geom_errorbarh(aes(xmax=abs+abs.SE, xmin=abs-abs.SE, y=emmean, color=color))+
 theme(legend.title = element_blank())+
  scale_color_manual(values=c(dot_colors[3], dot_colors[2]))+
  theme(legend.position = c(0.55, 0.15))+
  geom_text_repel(data=subset(emW, bac_name != "All 10 bacteria"), aes(x=abs, y=emmean, label=trimws(gsub("[\r\n]", "", bac_name))),fontface = "italic", size=2.5)+
      geom_text_repel(data=subset(emW, bac_name == "All 10 bacteria"), aes(x=abs, y=emmean, label=trimws(gsub("[\r\n]", "", bac_name))), size=2.5)+
  geom_smooth(aes(x=abs, y=emmean), method="lm", se=FALSE)+
  scale_y_continuous(limits=c(-0.6,0.6))+
  scale_x_continuous(limits=c(-2.5,2.5))+
    ggtitle("Wellspring")
plot_W_fitness

plot_2020_fitness <- plot_grid(plot_C_fitness, plot_W_fitness, labels="AUTO", align='h')
plot_2020_fitness
save_plot("Figure_6.pdf", plot_2020_fitness, base_height=6, base_width=12)

```
